### ISSUE TITLE
Generalize tx_limiter so non-transactional concurrent queries can also be limited

### ISSUE THREAD

**Issue by dweitzman** [vitessio/vitess#5113]
The transaction limiter supports limiting the number of active transactions at any given moment by caller ID.

Being able to limit the number of non-transactional concurrent queries by caller ID would be useful for the same reasons.

**Comment by pH14** [comment]
Agreed on this one -- we've had issues before where individual services were sending very expensive read queries that wound up blocking all of the MySQL connections. Seems like we could extend the `TxLimiter` to limit all connection acquisitions from the pool.
